File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int low=0;
        int high=arr.length-1;
        
        while(low<=high){
            int mid=(low+high)/2;

            if(arr[mid]-(mid+1)>=k)  // arr[mid]-(mid+1)  it gives how many missing no. in the left
                high=mid-1;
            else
                low=mid+1;
        }
        return low+k;
    }
}

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        maxNumber = max(arr)
        res = []
        for i in range(1,maxNumber+k+1):
            if i not in arr:
                res.append(i)
        for i in range(len(res)):
            print(res[i])
            if i==k-1:
                return res[i]
        return -1